### PR TITLE
LPD-36797 Quantity selector starting quantity is not correct when UOM is set with decimal base unit quantity

### PR DESCRIPTION
**Pull request by michelevigilante**
https://liferay.atlassian.net/browse/LPD-36797

### CODE REVIEW

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:playwright:nocompile

**Comment by michelevigilante** [comment]
ci:stop

**Comment by michelevigilante** [comment]
ci:stop

**Comment by michelevigilante** [comment]
ci:test:playwright

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:report:57115111

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:playwright

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:playwright

**Comment by michelevigilante** [comment]
ci:report:60052098

**Comment by michelevigilante** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/AddToCart.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ import {
 	CP_UNIT_OF_MEASURE_SELECTOR_CHANGED,
 } from '../../utilities/eventsDefinitions';
 import {useCommerceAccount, useCommerceCart} from '../../utilities/hooks';
-import {getMinQuantity} from '../../utilities/quantities';
+import {getMinQuantity, getMultiple} from '../../utilities/quantities';
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/add_to_cart/AddToCart.js:18]: Can we rename it to `getMultipleQuantity`?

File: modules/test/playwright/pages/commerce/commerce-product-content-web/productDetailsPage.ts [hunk 2]
```diff
@@ -183,6 +184,7 @@ export class ProductDetailsPage {
 		this.priceField = async (price: string, container = this.page) => {
 			return container.getByText(price);
 		};
+		this.productDetails = page.locator('.product-detail');
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-product-content-web/productDetailsPage.ts:187]: I think that this locator is too bound to the test. Either we remove it or we move the logic in the `quantitySelector` locator

File: modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumCatalogPage.ts [hunk 3]
```diff
@@ -21,9 +21,13 @@ export class CommerceThemeMiniumCatalogPage {
 		orderId: string,
 		accountName: string
 	) => Locator;
+	readonly quantitySelector: (targetLocator: Locator) => Locator;
 	readonly optionsButton: Locator;
 	readonly orderByButton: Locator;
 	readonly page: Page;
+	readonly popOverRules: (text: string) => Locator;
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumCatalogPage.ts:28]: Why is it plural?

Comment (alessiorendina) [reply modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumCatalogPage.ts:28]: Also, why the param is called `text` here?

File: modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumCatalogPage.ts [hunk 4]
```diff
@@ -62,14 +66,26 @@ export class CommerceThemeMiniumCatalogPage {
...
 				'[id^="portlet_com_liferay_commerce_product_content_search_web_internal_portlet_CPSortPortlet"]'
 			)
 			.getByTitle('Options');
 		this.orderByButton = page.locator('#commerce-order-by');
 		this.page = page;
+		this.popOverRules = (popOverRules: string) =>
+			this.page
+				.locator('.popover-body')
+				.getByText(popOverRules, {exact: true});
+		this.productCardItem = (productName: string) =>
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumCatalogPage.ts:82]: Is this used to locate a specific product card or a specific item inside a product card?

File: modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumCatalogPage.ts [hunk 5]
```diff
@@ -62,14 +66,26 @@ export class CommerceThemeMiniumCatalogPage {
...
+			this.page.locator('.product-card').filter({hasText: productName});
+		this.productCardItemAddToCartButton = (productName: string) =>
```
Comment (alessiorendina) [inline modules/test/playwright/pages/commerce/commerce-theme-minium/commerceThemeMiniumCatalogPage.ts:84]: Same here

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@michelevigilante please check my comments, thank you!

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:report:71211729

File: modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts [hunk 6]
```diff
@@ -86,3 +89,321 @@ test('LPD-33466 User can update pricing quantity of UOM', async ({
...
+		await expect(
+			commerceThemeMiniumCatalogPage.quantitySelector(
+				commerceMiniCartPage.miniCartItem(productName)
+			)
+		).toHaveValue('3');
+
+		await commerceThemeMiniumCatalogPage
+			.quantitySelector(commerceMiniCartPage.miniCartItem(productName))
+			.focus();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts:204]: Remove empty line

File: modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts [hunk 7]
```diff
@@ -86,3 +89,321 @@ test('LPD-33466 User can update pricing quantity of UOM', async ({
...
+		await expect(
+			commerceThemeMiniumCatalogPage.quantitySelector(
+				page.locator('.product-detail')
+			)
+		).toHaveValue('3');
+
+		await commerceThemeMiniumCatalogPage
+			.quantitySelector(page.locator('.product-detail'))
+			.focus();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts:232]: Same here

File: modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts [hunk 8]
```diff
@@ -86,3 +89,321 @@ test('LPD-33466 User can update pricing quantity of UOM', async ({
...
+		}
+	);
+
+	const productSkus = await apiHelpers.headlessCommerceAdminCatalog
+		.getProduct(product.productId)
+		.then((product) => {
+			return product.skus;
+		});
+
+	const sku = productSkus[0];
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts:319]: Reduce the number of calls here by using `const sku = product.skus[0]` instead.

File: modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts [hunk 9]
```diff
@@ -86,3 +89,321 @@ test('LPD-33466 User can update pricing quantity of UOM', async ({
...
+
+	const productName = product.name['en_US'];
+
+	const productSkus = await apiHelpers.headlessCommerceAdminCatalog
+		.getProduct(product.productId)
+		.then((product) => {
+			return product.skus;
+		});
+
+	const sku = productSkus[0];
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts:155]: Reduce the number of calls here by using `const sku = product.skus[0]` instead.

File: modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts [hunk 10]
```diff
@@ -86,3 +89,321 @@ test('LPD-33466 User can update pricing quantity of UOM', async ({
...
+			commerceThemeMiniumCatalogPage.productCard(productName)
+		)
+	).toHaveValue('1.2');
+
+	await commerceThemeMiniumCatalogPage
+		.quantitySelector(
+			commerceThemeMiniumCatalogPage.productCard(productName)
+		)
+		.focus();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts:345]: Remove empty line

File: modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts [hunk 11]
```diff
@@ -86,3 +89,321 @@ test('LPD-33466 User can update pricing quantity of UOM', async ({
...
+		await expect(
+			commerceThemeMiniumCatalogPage.quantitySelector(
+				commerceMiniCartPage.miniCartItem(productName)
+			)
+		).toHaveValue('1.2');
+
+		await commerceThemeMiniumCatalogPage
+			.quantitySelector(commerceMiniCartPage.miniCartItem(productName))
+			.focus();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts:368]: Same here

File: modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts [hunk 12]
```diff
@@ -86,3 +89,321 @@ test('LPD-33466 User can update pricing quantity of UOM', async ({
...
+		await expect(
+			commerceThemeMiniumCatalogPage.quantitySelector(
+				page.locator('.product-detail')
+			)
+		).toHaveValue('1.2');
+
+		await commerceThemeMiniumCatalogPage
+			.quantitySelector(page.locator('.product-detail'))
+			.focus();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts:396]: Same here

File: modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts [hunk 13]
```diff
@@ -86,3 +89,321 @@ test('LPD-33466 User can update pricing quantity of UOM', async ({
...
+			commerceThemeMiniumCatalogPage.productCard(productName)
+		)
+	).toHaveValue('3');
+
+	await commerceThemeMiniumCatalogPage
+		.quantitySelector(
+			commerceThemeMiniumCatalogPage.productCard(productName)
+		)
+		.focus();
+
```
Comment (alessiorendina) [inline modules/test/playwright/tests/commerce/commerce-product-definitions-web/skuUnitOfMeasure.spec.ts:181]: Same here

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@michelevigilante LGTM overall, just few small changes. Thank you!

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:playwright

**Comment by michelevigilante** [comment]
ci:report:72282797

**Pull request closed without merging**